src/controllers/gamepads.jsx

```diff
@@ -0,0 +1,171 @@
+const MAX_DRIVE_ANGLE = 12;
+const MAX_TRANSLATE_ANGLE = 45;
+
+const MIN_ROTUNDA_ANGLE = -90;
+const MAX_ROTUNDA_ANGLE = 90;
+
+const MIN_SHOULDER_ANGLE = 0;
+const MAX_SHOULDER_ANGLE = 90;
+
+const MIN_ELBOW_ANGLE = -180;
+const MAX_ELBOW_ANGLE = 180;
+
+const MIN_WRIST_PITCH_ANGLE = -90;
+const MAX_WRIST_PITCH_ANGLE = 90;
+
+const MIN_WRIST_ROLL_ANGLE = -360;
+const MAX_WRIST_ROLL_ANGLE = 360;
+
+const MIN_END_EFFECTOR_ANGLE = 0;
+const MAX_END_EFFECTOR_ANGLE = 150;
+
+function clamp(num, min, max) {
+    return Math.min(Math.max(num, min), max);
+}
```

Comment:
Might want to centralize everything above this line to a more generic source file, since it can apply in many parts of the code